### DISCUSSION TITLE
Successful Recovery returns 422 instead of 200

### DISCUSSION THREAD

**Discussion by Big-Brother-Man** [ory/.github#53] [Q&A] [answered]
Hi all, I hope you are all doing well.

When I have a successful recovery of a user instead of getting 200 I get a 422. Here is a snpitted of the response

```
       redirect_browser_to: "/ui/settings?flow=245004c8-daae-4258-85d3-8c6ad63bf56a"
            error: {
              "id": "browser_location_change_required",
              "code": 422,
              "status": "Unprocessable Entity",
              "reason": "In order to complete this flow please redirect the browser to: /ui/settings?flow=245002d8-daae-5158-85d3-8c1af63bf56a",
              "message": "browser location change required"
            }
```
Is this intended? Please let me know (maybe is best possible response duo to a redirect?) and explain to me like I am stupid thank you!

**Comment by Benehiko** [comment, marked as answer]
Hi @Big-Brother-Man 

When working with AJAX the response is 422 when a new flow needs to be created. In server-side Apps a redirect will occur automatically.

This happens in submitting oidc and passwordless flows, among other instances. 

https://www.ory.sh/docs/kratos/bring-your-own-ui/custom-ui-advanced-integration#spas-and-the-422-error

**Reply by Big-Brother-Man** [reply]
Hey @Benehiko many thanks if I understand it correct in server side we utilise the payload to extract the flow and redirect the user to the given route. But in client the redirect is taken for you?

**Reply by Benehiko** [reply]
Hey @Big-Brother-Man 

It depends on how your application (server or client side) is calling the Ory endpoint. Please take a look at this document https://www.ory.sh/docs/kratos/bring-your-own-ui/custom-ui-basic-integration#browser-vs-native-apps.

When you are relying on browser redirects with a server-side rendered app such as Express.Js then the redirect will happen automatically since you are natively posting the values to Ory directly. But if you are using the SDK to post the values through JSON then it will give you a 422 which you need to handle.

**Reply by Big-Brother-Man** [reply]
Many thanks for clarification that makes a whole lot of sense. I had SDK thus just got the response back without no redirect. Thank you.

**Comment by LandSprutte** [comment]
The documentation you link to assume, you are getting a response including a url with a flowId.

But I receive a oidc redirect looking more like:  "http://localhost:4000/oauth2/auth?audience=..."

How do I handle this flow?
